crates/oxc_semantic/src/builder.rs

```diff
@@ -203,10 +206,23 @@ impl<'a> SemanticBuilder<'a> {
     }
 
     fn create_ast_node(&mut self, kind: AstKind<'a>) {
+        let span = kind.span();
+
+        let mut leading_comments = vec![];
+        // Should handle trailing comments as well?
+        for (start, comment) in self.trivias.comments().range(..span.start) {
+            if !self.comments_before_seen.contains(start) {
+                leading_comments.push((start, comment));
+            }
+            self.comments_before_seen.insert(*start);
+        }
```

Comment:
I think these logic belong to the jsdoc builder, having implementation details here not clean.

Reply:
The reason why I keep this here was,
- there are various usecases for comments
  - such as `@ts-ignore`, `eslint-ignore` and `prettier-ignore`, etc..
- but finding can be done at once in advance

How do you think?

Reply:
As long as things are well encapsulated, it's going to be hard to navigate the code since this file is huge with so many builders ...

Reply:
OK, I'll move these to jsdoc builder.

(BTW, I wish the AST included comments... 😴 )